utilities/inference_utils.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+import json
+import re
+import shlex
+from functools import cache
+from json import JSONDecodeError
+from string import Template
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from ocp_resources.inference_service import InferenceService
+from pyhelper_utils.shell import run_command
+from simple_logger.logger import get_logger
+
+from utilities.manifests.runtime_query_config import RUNTIMES_QUERY_CONFIG
+
+LOGGER = get_logger(name=__name__)
+INFERENCE_QUERIES: Dict[str, Dict[str, str]] = {
+    "nitrogen-boil-temp": {
+        "text": "At what temperature does liquid Nitrogen boil?",
+        "response_text": "74 degrees F",
+    }
+}
+
+
+class Inference:
+    ALL_TOKENS: str = "all-tokens"
+    STREAMING: str = "streaming"
+
+    def __init__(self, inference_service: InferenceService, runtime: str, protocol: str, inference_type: str):
+        """
+        Args:
+            inference_service: InferenceService object
+        """
+        self.inference_service = inference_service
+        self.runtime = runtime
+        self.protocol = protocol
+        self.inference_type = inference_type
+
+    @cache
+    def get_inference_url(self) -> str:
+        if url := self.inference_service.instance.status.components.predictor.url:
```

Comment:
I think this is different if the inference service uses ModelMesh instead of KServe. @lugi0 can you confirm? If you are not gonna use ModelMesh for now, maybe we can address this on a different PR.

Reply:
added a `TODO` to support modelmesh